x-pack/plugins/ml/public/application/timeseriesexplorer/timeseriesexplorer_utils/timeseries_search_service_provider.ts

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { each, find, get, filter } from 'lodash';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import type { MlEntityField } from '@kbn/ml-anomaly-utils';
+import type { Job } from '../../../../common/types/anomaly_detection_jobs';
+import type { ModelPlotOutput } from '../../services/results_service/result_service_rx';
+import type { MlApiServices } from '../../services/ml_api_service';
+import type { MlResultsService } from '../../services/results_service';
+import { buildConfigFromDetector } from '../../util/chart_config_builder';
+import {
+  isModelPlotChartableForDetector,
+  isModelPlotEnabled,
+} from '../../../../common/util/job_utils';
+
+export function timeSeriesSearchServiceProvider(
```

Comment:
Suggest to rename this to `timeSeriesSearchServiceFactory` and the file to `time_series_search_service.ts` to be in line with how we renamed the time buckets service.

Reply:
@walterra - all naming updates made in 659f1ca4a4362837692cfd38cb69053f092bc26a